### PR TITLE
Refactor and test the builds:index command

### PR DESCRIPTION
**Pull request by dmathieu**
* Instead of using console.log, we use cli.table.
* Handle limiting the number of builds returned too.

### CODE REVIEW

**Comment by dmathieu** [comment]
ping @ransombriggs @dickeyxxx for review

File: package.json [hunk 1]
```diff
@@ -25,7 +25,8 @@
     "heroku-cli-util": "5.10.10",
     "ignore": "3.1.2",
     "node-uuid": "1.4.7",
-    "request": "2.72.0"
+    "request": "2.72.0",
```
Comment (dickeyxxx) [inline package.json]: not part of this PR, but we discourage use of this package

File: package.json [hunk 2]
```diff
@@ -25,7 +25,8 @@
     "heroku-cli-util": "5.10.10",
```
Comment (dickeyxxx) [inline package.json]: this is pretty far out of date, should update

File: test/commands/builds/index.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
+'use strict'
+/* globals describe beforeEach it */
```
Comment (dickeyxxx) [inline test/commands/builds/index.js]: you can avoid this by setting `"env": "mocha"` for standard in package.json

File: package.json (hunk 1 above)
Comment (dmathieu) [inline package.json]: Thank you, you're right. I'll remove it.
I'd rather keep this PR focused on the tests though.

File: package.json (hunk 2 above)
Comment (dmathieu) [inline package.json]: :+1: same as previous comment.

File: test/commands/builds/index.js (hunk 3 above)
Comment (dmathieu) [inline test/commands/builds/index.js]: I actually copied this test from another suite, and it appears this comment is entirely unnecessary, without any need for other changes. I've removed it.

**Comment by dmathieu** [comment]
:+1: for the `--json` flag. I'm mostly just trying to improve the maintainability of this tool at this point though (and doing that on the side, so trying to keep PRs light).
